### PR TITLE
Layout secondary devtools view properly

### PR DESCRIPTION
**Pull request by sangwoo108**
Set the resizing strategy for the secondary devtools view and update it during layout.

Resolves https://github.com/brave/brave-browser/issues/37413

image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by sangwoo108** [comment]
friendly reminder 😀

File: browser/ui/views/frame/brave_browser_view.h [hunk 1]
```diff
@@ -184,6 +184,10 @@ class BraveBrowserView : public BrowserView,
   bool IsActiveWebContentsTiled(const SplitViewBrowserData::Tile& tile) const;
   void UpdateSecondaryContentsWebViewVisibility();
 
+  // Returns the content::WebContents* for devtools.
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.h:187]: Maybe need to update this comment?

File: browser/ui/views/frame/brave_browser_view.cc [hunk 2]
```diff
@@ -406,18 +407,44 @@ void BraveBrowserView::UpdateSecondaryContentsWebViewVisibility() {
...
+  DevToolsContentsResizingStrategy strategy;
+  content::WebContents* devtools =
+      DevToolsWindow::GetInTabWebContents(inspected_contents, &strategy);
+  if (secondary_devtools_web_view_->web_contents() != devtools) {
+    secondary_devtools_web_view_->SetWebContents(devtools);
+  }
+
+  if (secondary_devtools_web_view_->web_contents() != devtools) {
+    secondary_devtools_web_view_->SetWebContents(devtools);
+  }
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.cc:436]: can be deleted. c&p mistake?

File: browser/ui/views/frame/brave_browser_view.cc [hunk 3]
```diff
@@ -406,18 +407,44 @@ void BraveBrowserView::UpdateSecondaryContentsWebViewVisibility() {
...
+    UpdateSecondaryDevtoolsLayoutAndVisibility(contents);
+
     auto* contents_layout_manager = static_cast<BraveContentsLayoutManager*>(
         contents_container()->GetLayoutManager());
     contents_layout_manager->show_main_web_contents_at_tail(
         second_tile_is_active_web_contents);
   } else {
     secondary_contents_web_view_->SetWebContents(nullptr);
     secondary_contents_web_view_->SetVisible(false);
+    secondary_devtools_web_view_->SetVisible(false);
```
Comment (simonhong) [inline browser/ui/views/frame/brave_browser_view.cc:419]: don't need this here? `secondary_devtools_web_view_->SetWebContents(nullptr);`

**Review by simonhong — COMMENTED** [review]
sorry for late reply! It's slipped through the crack.

File: browser/ui/views/frame/brave_browser_view.h (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/frame/brave_browser_view.h:187]: Oops, right. We don't need the comment anymore.

File: browser/ui/views/frame/brave_browser_view.cc (hunk 2 above)
Comment (sangwoo108) [reply browser/ui/views/frame/brave_browser_view.cc:436]: Yeah, should be gone

File: browser/ui/views/frame/brave_browser_view.cc (hunk 3 above)
Comment (sangwoo108) [reply browser/ui/views/frame/brave_browser_view.cc:419]: That'd be better.

**Review by simonhong — APPROVED** [review]
👍🏼

**Pull request merged**